Bugfix: Prevent Error 500 and Update README.md mentioning new dependencies and pitfalls after merge of "feature boxplot-taskstatistic"

ifrh opened the pull request:
@ratefuchs : Most of changes in 25 files are orthography fixes. But there are some functional fixes to.

1. Insert information about new dependencies introduced by merging "feature boxplot-taskstatistic" ( https://github.com/KITPraktomatTeam/Praktomat/pull/345 )  .
2. Update README.md to give more information about installation steps : Fixes https://github.com/KITPraktomatTeam/Praktomat/issues/250 .
3. Fix some orthography errors in multiple files.
4. To let GitHub-Actions successfully test Praktomat-Code , I merged into hbrs_public changes from 

- @hannesbraun: https://github.com/KITPraktomatTeam/Praktomat/pull/351 , 
- @hannesbraun: https://github.com/KITPraktomatTeam/Praktomat/pull/349 , 
- @hannesbraun: https://github.com/KITPraktomatTeam/Praktomat/pull/348

5. Repair some how broken commit-history depending on former hbrs_changes and newer hbrs_public  branch, but no current file change has its reason in this point.

\[ Edit : 6.9.2022: 
6. Additional commit a bugfix to prevent an Error 500 while running TaskAdmin.run_all_uploadtime_checkers_on_all, see https://github.com/ifrh/Praktomat/commit/af7cff11cd528ad7768ca43168cd219d35985fb3 \]

Commit:
HBRS 2013 before Wintersemester, test VM

Commit:
HBRS 2015 Wintersemester, BIS, production

Commit:
HBRS 2013 Wintersemester, production

Commit:
HBRS 2015 Wintersemester, BCS, production

Commit:
HBRS 2014 Wintersemester, production

Commit:
HBRS 2015 , admin-tutor-test

Commit:
HBRS 2016 Sommersemester, test

Commit:
HBRS 2014 Sommersemester, production

Commit:
HBRS 2015 Wintersemester, BCS-pract, production

Commit:
HBRS 2015 Sommersemester, production

Commit:
make shure submodules can be recieved via http_proxy ...

Signed-off-by: Robert Hartmann (H-BRS,FB02) <Robert.Hartmann@h-brs.de>

Commit:
Merge remote-tracking branch 'HBRS_2013w/master' into HBRS_Merge_Local

Conflicts:
	setup.py
	src/checker/checker/__init__.py

choose 'HBRS_2013w/master'

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge remote-tracking branch 'HBRS_2013w/HBRS_2013w_production' into HBRS_Merge_Local

Conflicts:
	src/checker/checker/DiffChecker.py
	src/checker/checker/__init__.py
	src/settings_local.py

using versions from 'HBRS_2013w/HBRS_2013w_production' with small modifications

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge remote-tracking branch 'HBRS_2014s/HBRS_2014s' into HBRS_Merge_Local

Conflicts:
	src/accounts/views.py
	src/checker/checker/AutoAttestChecker.py
	src/checker/checker/DiffChecker.py
	src/settings_local.py

using 'HBRS_2014s/HBRS_2014s' with small modifications

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge remote-tracking branch 'HBRS_2014w/HBRS_2014w' into HBRS_Merge_Local

Conflicts:
	src/accounts/views.py
	src/checker/checker/AutoAttestChecker.py
	src/checker/checker/DiffChecker.py
	src/settings_local.py

using 'HBRS_2014w/HBRS_2014w' with small modifications

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge remote-tracking branch 'HBRS_2014_15_test/HBRS_2014_15_test' into HBRS_Merge_Local

Conflicts:
	src/accounts/views.py
	src/checker/checker/AutoAttestChecker.py
	src/checker/checker/DiffChecker.py
	src/checker/scripts/javac
	src/settings_local.py

using 'HBRS_2014_15_test/HBRS_2014_15_test' with small modifications

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge remote-tracking branch 'HBRS_2015s/HBRS_2015s' into HBRS_Merge_Local

Conflicts:
	src/accounts/views.py
	src/checker/checker/AutoAttestChecker.py
	src/checker/checker/DiffChecker.py
	src/checker/scripts/javac
	src/settings_local.py

using 'HBRS_2015s/HBRS_2015s' with small modifications

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge remote-tracking branch 'HBRS_2015w-bcs/HBRS_2015w-bcs' into HBRS_Merge_Local

Conflicts:
	src/accounts/views.py
	src/checker/checker/AutoAttestChecker.py
	src/checker/checker/DiffChecker.py
	src/checker/scripts/java.policy
	src/checker/scripts/junit.policy
	src/settings_local.py

using 'HBRS_2015w-bcs/HBRS_2015w-bcs' with small modifications ...

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge remote-tracking branch 'HBRS_2015w-bcs-practice/HBRS_2014w' into HBRS_Merge_Local

Conflicts:
	src/accounts/views.py
	src/checker/checker/AutoAttestChecker.py
	src/checker/checker/DiffChecker.py
	src/checker/scripts/junit.policy
	src/settings_local.py

using 'HBRS_2015w-bcs-practice' with small modifications

Commit:
Merge remote-tracking branch 'HBRS_2015w-bis/HBRS_2015w-bis' into HBRS_Merge_Local

Conflicts:
	src/accounts/views.py
	src/checker/checker/AutoAttestChecker.py
	src/checker/checker/DiffChecker.py
	src/checker/scripts/junit.policy
	src/settings_local.py

using 'HBRS_2015w-bis/HBRS_2015w-bis' with small modifications

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge remote-tracking branch 'HBRS_2016s/HBRS' into HBRS_Merge_Local

Conflicts:
	.gitmodules

changes url for ace-submodule from  git: via http: to https: to become http_proxy compatible ...

Commit:
Merge remote-tracking branch 'HBRS_2016s/HBRS_s2016' into HBRS_Merge_Local

Conflicts:
	.gitmodules
	src/accounts/admin.py
	src/accounts/urls.py
	src/accounts/views.py
	src/checker/checker/DiffChecker.py
	src/checker/checker/JUnitChecker.py
	src/checker/checker/__init__.py
	src/checker/compiler/JavaBuilder.py
	src/checker/compiler/__init__.py
	src/checker/models.py
	src/checker/scripts/java.policy
	src/checker/scripts/javac
	src/settings.py
	src/settings_local.py
	src/tasks/views.py
	src/templates/500.html
	src/templates/base.html
	src/templates/registration/login.html
	src/templates/tasks/task_list.html

Commit:
reinsert LDAP-users and handling with a quixfix (is it a fix?)
# FIXME - see KITPraktomat Bugreport nr 221
# https://github.com/KITPraktomatTeam/Praktomat/issues/221

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
solving "runtime merge conflicts..."
Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
solving "runtime merge conflicts... remove old SOUTH-Migration files and insert newer DJANGO-Migration files"
Thanks to Stephan :-)

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
configuration in src/settings/defaults.py,  src/settings/devel.py, src/settings/local.py, src/settings/test.py

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Attestation_view-Template : Allow Tutor+Trainer to watch solutionfiles and not only allow to download them

Commit:
Resubmit Solution => Java Compiler failed <=
If new SolutionNumber has more position than the old SolutionNumber
the filename in Sandbox got damaged: i.e:
SolutionArchive/[...]/Solution_9993/AlgoSchlange.java
SolutionSandbox/[...]/Solution_10039/lgoSchlange.java

Solved by using regular-expression in path-method in solution/models.py

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge of current HBRS-settings from old Praktomat versions ...

Commit:
move and rename old hbrs praktomat settings file

Commit:
Attestation_view-Template : Allow Tutor+Trainer to watch solutionfiles and not only allow to download them
solving Template runtime error introduced by merging our changes from
Praktomat 2013 into upstream Praktomat 2016

Commit:
AutoAttest-Checker: made Praktomat2016 compatible, changed default values

Commit:
AutoAttest-Checker: repair setting final solutions and repair saving attestations

Commit:
Task-Admin: to prevend "repair functions" doing from dbshell and filesystem
            implemented missing functions
           - run_all_checkers_on_latest_only_failed
           - run_all_checker => run_all_checkers_on_finals
           - delete_attestations
           - run_all_uploadtime_checkers_on_all (usefull after some submissions used checkers hat to be changed)
           - unset_all_checker_finished

Commit:
Tasks: Modell + Solution: View // implement requirements for limiting uploads or/and waiting time between possible uploads

Commit:
Task-Admin: missing html-template used in function run_all_uploadtime_checkers_on_all

Commit:
individualised Settings (devel.py + local.py) , remove old backup settingsfile

Commit:
AutoAttestChecker: migration py

Commit:
settings/default.py : bug in class D, could not save values as attributes ...

Commit:
remove 404 Http-Error for favico.ico

Commit:
Autoattest-Checker: Add ValidationError

Commit:
settings/devel.py add comment for not using PRAKTOMAT_ID with runserver, runserver_plus

Commit:
Merge branch 'master' of https://github.com/KITPraktomatTeam/Praktomat.git into HBRS_Merge_Local
into hbrs_local_merge
Conflicts:
	docker-image/Dockerfile
	src/accounts/forms.py
	src/context_processors/settings.py
	src/settings/devel.py
	src/settings/local.py
	src/templates/attestation/rating_export.csv

Commit:
Migration files merge HBRS

Commit:
Fix: django-debug-toolbar + sqlparse
http://stackoverflow.com/questions/38479063/django-debug-toolbar-breaking-on-admin-while-getting-sql-stats

Commit:
hbrs: make compatible diffchecker

Commit:
JPLAGJAR in /opt/praktomat-addons/jplag.jar

Commit:
Dockerfile: Insert line "RUN pip3 install --upgrade pip setuptools wheel"

Commit:
Dockerfile: comment out lines for Isabelle support
       ... they break at least here (HBRS) the
           dockerimage-creation  ...
we don't use Isabelle-Checker, so it doesn't matter for us.

Commit:
Small code layout modifications ...

Commit:
src/settings/local.py correct for HBRS deployment needs ...

Commit:
Merge branch 'master' of https://github.com/KITPraktomatTeam/Praktomat.git into HBRS_Merge_Local
only requirements.txt

ifrh committed:
Update AutoAttestChecker.py

ifrh committed:
Update 0003_merge-hbrs.py

There it is a known django-bug, that "Removing a model's last field results in SQL syntax error" strongly dependend on the used database management system https://code.djangoproject.com/ticket/24424 - the solution for the sqlite-backend would be changing django/db/backends/sqlite3/schema.py on the travis-server like https://github.com/django/django/pull/4325/files#diff-0c8f495bfee773ab7b5409533bd6d7efL142 shows.
Just to prevent this bug for Travis-CI Test (using sqlite backend) 
the solution is, that migration-files hat to be changed to get working with sqlite.
Here concrete it is in: src/checker/migrations/0003_merge-hbrs.py remove (or comment out)
all RemoveField-Operation for corresponding DeleteModel-Operations.

Commit:
This fixes
https://github.com/KITPraktomatTeam/Praktomat/issues/255
#255 "Praktomat Timeout didn't work with every kind of checkers. <== Includes Patch for Praktomat/src/utilities/safeexec.py"
This fix has been used at H-BRS at semester WiSe 2016/17 for testing.
Works for Ubuntu 12.04.5 LTS and Ubuntu 14.04.4 LTS.
Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
JUNIT-Policy cleanup but
make settings compatible for JUNIT-PARAMS

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Tasks: students upload waitingtime changed from linear time-function to constant time.

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
some HBRS-specific cleanups for src/settings/local.py

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
correct some wording in template file:
src/templates/solutions/solution_list.html

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
change file mode for src/urls.py

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Revert like merge branch 'master' into 'HBRS_Merge_Local'
master differs from HBRS_Merge_Local,
removed differences by using checkout master files:
M	.gitmodules
M	.travis.yml
M	README.md
M	docker-image/Dockerfile
D	docker-image/Rprofile
A	examples/docker/R/Dockerfile
A	examples/docker/R/Rprofile
M	media/styles/style.css
M	requirements.txt
M	src/accounts/admin.py
M	src/accounts/forms.py
D	src/accounts/ldap_auth.py
M	src/accounts/middleware.py
M	src/accounts/models.py
M	src/accounts/urls.py
M	src/accounts/views.py
M	src/checker/checker/AnonymityChecker.py
D	src/checker/checker/AutoAttestChecker.py
M	src/checker/checker/CheckStyleChecker.py
M	src/checker/checker/CreateFileChecker.py
M	src/checker/checker/DejaGnu.py
D	src/checker/checker/DiffChecker.py
M	src/checker/checker/InterfaceChecker.py
M	src/checker/checker/IsabelleChecker.py
M	src/checker/checker/JUnitChecker.py
M	src/checker/checker/LineCounter.py
M	src/checker/checker/LineWidthChecker.py
M	src/checker/checker/ScriptChecker.py
M	src/checker/checker/TextChecker.py
M	src/checker/checker/__init__.py
M	src/checker/compiler/Builder.py
M	src/checker/compiler/JavaBuilder.py
A	src/checker/compiler/ScalaBuilder.py
M	src/checker/compiler/__init__.py
D	src/checker/migrations/0003_merge-hbrs.py
D	src/checker/migrations/0004_hbrs_autoattestchecker.py
A	src/checker/migrations/0004_textchecker_and_scalabuilder.py
D	src/checker/migrations/0005_merge_hbrs_jplag.py
D	src/checker/migrations/0006_hbrs_diffchecker.py
M	src/checker/scripts/java
M	src/checker/scripts/java.policy
M	src/checker/scripts/javac
M	src/checker/scripts/junit.policy
D	src/checker/scripts/junit.policy.old
D	src/checker/scripts/junit.policy.old2
M	src/context_processors/settings.py
M	src/settings/defaults.py
M	src/settings/devel.py
M	src/settings/local.py
M	src/settings/test.py
M	src/solutions/models.py
D	src/solutions/templatetags/tasks_extras.py
M	src/solutions/views.py
M	src/tasks/admin.py
D	src/tasks/migrations/0002_merge-hbrs.py
D	src/tasks/migrations/0003_hbrs_submission_limits.py
D	src/tasks/migrations/0004_merge_hbrs_jplag.py
M	src/tasks/models.py
M	src/tasks/views.py
M	src/templates/admin/accounts/user/change_list.html
D	src/templates/admin/accounts/user/import_ldap.html
M	src/templates/admin/tasks/task/change_form.html
M	src/templates/attestation/attestation_view.html
M	src/templates/attestation/rating_export.csv
M	src/templates/attestation/rating_overview.html
M	src/templates/base.html
A	src/templates/checker/compiler/scala_builder_report.html
M	src/templates/registration/login.html
M	src/templates/registration/registration_change.html
A	src/templates/registration/registration_deactivated.html
M	src/templates/solutions/solution_detail.html
M	src/templates/solutions/solution_list.html
D	src/templates/solutions/submission_final_lost_email.html
M	src/templates/tasks/task_detail.html
M	src/templates/tasks/task_list.html
M	src/urls.py
M	src/utilities/safeexec.py

Now this branch has the same files like KITPraktomatTeams master.
this commit is the new starting commit for hbrs-features in own branches like
- "git-submodules via https"
- "AutoAttestChecker"
- "Dynamic upload waiting time"
- "Praktomat - LDAP connection"

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge branch 'feature_gitmodules_https-proxy_compatible' into 'hbrs_public'

closes Feature-Request: Make .gitmodules compatible with https-proxys

See merge request !3

Commit:
Merge branch 'bugfix_safeexec' into 'hbrs_public'

Bugfix safeexec

See merge request !5

Commit:
Merge branch 'feature_diffchecker' into 'hbrs_public'

DiffChecker reintroduce a scriptbased DiffChecker

See merge request !6

Commit:
Merge branch 'feature_autoattestchecker' into 'hbrs_public'

HBRS-Feature AutoAttestChecker

See merge request !7

Commit:
Merge branch 'feature_admin-additional-actions_avoid_working_via_dbshell' into hbrs_public

resolved Conflicts:
	src/tasks/admin.py

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge branch 'bugfix_solution_resubmit' into 'hbrs_public'

Bugfix for issue #232 (resubmit bug)

See merge request !9

Commit:
Merge branch 'feature_TrainerTutor_Watch_SolutionFiles' into 'hbrs_public'

Attestation_view-Template:

See merge request !10

Commit:
Merge branch 'bugfix_small_typos' into 'hbrs_public'

Bugfix small typos

See merge request !11

Commit:
Merge branch 'feature_dynamic_upload_waiting_time' into 'hbrs_public'

Feature: uploaders dynamic waiting time and max upload limit

See merge request !12

Commit:
Merge branch 'feature_sort_tasklist_via_names' into 'hbrs_public'

Feature: sort tasklist via task title and not only after submission date

See merge request !13

Commit:
Merge branch 'feature_TaskAdmin_Button_Modell_Solution' into 'hbrs_public'

TaskAdmin : Add upper site button for adding model solution

See merge request !14

Commit:
Merge branch 'feature_LDAP_bindings' into hbrs_public

resolved Conflicts:
	requirements.txt

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge branch 'feature_download_solutiondetails' into 'hbrs_public'

Add download link to Template solution_detail.html

See merge request !16

Commit:
Merge branch 'feature_disable_contactlink' into hbrs_public

resolved Conflicts:
	src/settings/defaults.py

Signed-off-by: Robert Hartmann (FB02, H-BRS) <Robert.Hartmann@h-brs.de>

Commit:
Merge branch 'feature_LDAP_bindings' into 'hbrs_public'

Update README.md

See merge request rhartm2m/praktomat_hbrs!19

Commit:
Isabelle checker : Add created Migrationfile missing in KITPraktomatTeam/master

Commit:
Merge branches 'master', 'fix_readme_pip_problem', 'fix_requirement_Django_1-11-15', 'bugfix_safeexec', 'bugfix_small_typos', 'bugfix_solution_resubmit', 'feature_TaskAdmin_Button_Modell_Solution', 'feature_TrainerTutor_Watch_SolutionFiles', 'feature_admin-additional-actions_avoid_working_via_dbshell', 'feature_disable_contactlink', 'feature_download_solutiondetails', 'feature_dynamic_upload_waiting_time', 'feature_gitmodules_https-proxy_compatible' and 'feature_sort_tasklist_via_names' into hbrsNov2018

Commit:
Merge branch 'feature_LDAP_bindings' into hbrsNov2018

# manual resolved Conflicts:
#	requirements.txt
#	src/settings/defaults.py

Commit:
Merge branch 'feature_CUnitCppUnit_Checker' into hbrsNov2018

Commit:
Merge branch 'feature_autoattestchecker' into hbrsNov2018

# manual resolved Conflicts:
#	src/checker/checker/__init__.py
#	src/tasks/admin.py

Commit:
Merge branch 'feature_checkstyleChecker_updVersion_8.14' into hbrsNov2018

Commit:
Merge branch 'feature_diffchecker' into hbrsNov2018

# manual resolved Conflicts:
#	src/checker/checker/__init__.py

Commit:
create migrationfile for merging HBRS migration-leafs

Commit:
Merge branch 'hbrsNov2018' into hbrs_public

# manual resolved Conflicts:
#	requirements.txt
#	src/settings/defaults.py
#	src/tasks/admin.py

Commit:
repair Checker migration files after completing merging

Commit:
Merge branch 'master' into fix_IsabelleChecker_Migrationfile

Commit:
Merge branch 'master' into hbrs_public

# Conflicts:
#	requirements.txt
#	src/accounts/urls.py

Commit:
Update src/checker/checker/AnonymityChecker.py, src/checker/checker/CheckStyleChecker.py, src/checker/checker/CreateFileChecker.py, src/checker/checker/DejaGnu.py, src/checker/checker/HaskellTestFrameWorkChecker.py, src/checker/checker/InterfaceChecker.py, src/checker/checker/IsabelleChecker.py, src/checker/checker/JUnitChecker.py, src/checker/checker/JavaChecker.py, src/checker/checker/KeepFileChecker.py, src/checker/checker/LineCounter.py, src/checker/checker/LineWidthChecker.py, src/checker/checker/RChecker.py, src/checker/checker/ScriptChecker.py, src/checker/checker/TextChecker.py, src/checker/compiler/Builder.py, src/checker/compiler/CBuilder.py, src/checker/compiler/CXXBuilder.py, src/checker/compiler/FortranBuilder.py, src/checker/compiler/HaskellBuilder.py, src/checker/compiler/JavaBuilder.py, src/checker/compiler/JavaGCCBuilder.py, src/checker/compiler/ScalaBuilder.py files

Commit:
Update .gitlab-ci.yml

Commit:
Update .gitlab-ci.yml

Commit:
Merge branch 'master32' into java-checker

Conflicts:
	src/checker/compiler/CBuilder.py
	src/checker/compiler/CXXBuilder.py
	src/checker/compiler/FortranBuilder.py
	src/checker/compiler/JavaGCCBuilder.py
	src/checker/compiler/ScalaBuilder.py
	src/settings/defaults.py

Commit:
fix Django migration-file for javaChecker: on_delete at ForeignKey for python3

Commit:
remove gitlab-ci.yml from branch

Commit:
Merge branches 'bugfix_safeexec', 'bugfix_small_typos' and 'bugfix_solution_resubmit' into octupus_merge_2020

Commit:
Merge remote-tracking branches 'origin/feature_gitmodules_https-proxy_compatible', 'origin/feature_TaskAdmin_Button_Modell_Solution', 'origin/feature_TrainerTutor_Watch_SolutionFiles', 'origin/feature_admin-additional-actions_avoid_working_via_dbshell', 'origin/feature_disable_contactlink', 'origin/feature_download_solutiondetails', 'origin/feature_dynamic_upload_waiting_time' and 'origin/feature_sort_tasklist_via_names' into octupus_merge_2020

Commit:
Merge remote-tracking branches 'origin/feature_CUnitCppUnit_Checker' and 'origin/java-checker' into octupus_merge_2020

Conflicts:
	src/checker/checker/__init__.py

Commit:
Merge remote-tracking branch 'origin/feature_autoattestchecker' into octupus_merge_2020

Conflicts:
	src/checker/checker/__init__.py

Commit:
Merge remote-tracking branch 'origin/feature_checkstyleChecker_updVersion_8.14' into octupus_merge_2020

Commit:
Merge remote-tracking branch 'origin/feature_diffchecker' into octupus_merge_2020

Conflicts:
	src/checker/checker/__init__.py

Commit:
Merge remote-tracking branch 'origin/fix_IsabelleChecker_Migrationfile' into octupus_merge_2020

Commit:
Merge branch 'feature_LDAP_bindings' into octupus_merge_2020

Conflicts:
	src/settings/defaults.py

Commit:
add Django migrationfiles for HBRS octopus branch merges. Python2 and Python3 compatible

Commit:
Merge branch 'octupus_merge_2020' into hbrs_public

Conflicts:
	requirements.txt
	src/accounts/urls.py
	src/checker/checker/__init__.py
	src/settings/defaults.py
	src/tasks/admin.py

Commit:
add Django migration mergefile

Commit:
Merge branch 'feature_LDAP_bindings' into hbrs_public

Conflicts:
	requirements.txt

Commit:
Merge branch 'feature_LDAP_bindings' into hbrs_public

Commit:
Merge branch 'master32' updated from 'master' into java-checker

Commit:
Merge branch 'master32' into hbrs_public

# Conflicts:
#	src/accounts/urls.py
#	src/accounts/views.py

Commit:
Merge branches 'feature_gitmodules_https-proxy_compatible', 'fix_readme_requirements', 'bugfix_small_typos', 'feature_TrainerTutor_Watch_SolutionFiles', 'bugfix_solution_resubmit', 'feature_admin-additional-actions_avoid_working_via_dbshell', 'feature_dynamic_upload_waiting_time', 'feature_autoattestchecker', 'feature_diffchecker', 'feature_CUnitCppUnit_Checker', 'feature_sort_tasklist_via_names' and 'java-checker' into hbrs_public

Commit:
Merge branch 'feature_LDAP_bindings' into hbrs_public

Commit:
Merge branch 'feature_checkstyleChecker_updVersion_8.14' into hbrs_public

Commit:
Merge branch 'feature_TaskAdmin_Button_Modell_Solution' into hbrs_public

Commit:
Merge branch 'feature_disable_contactlink' into hbrs_public

Commit:
Merge branch 'feature_download_solutiondetails' into hbrs_public

Commit:
Merge branch 'bugfix_safeexec' into hbrs_public

Commit:
Merge branch 'master32' updated from master into hbrs_public

Conflicts:
	src/templates/checker/compiler/builder_report.html

Commit:
Merge branch 'master32' updated from 'master' into java-checker

Commit:
Merge branch 'master32' updated from 'master' into hbrs_public

Commit:
Merge H-BRS feature und bugfix branches 'feature_gitmodules_https-proxy_compatible', 'bugfix_small_typos', 'feature_TrainerTutor_Watch_SolutionFiles', 'bugfix_solution_resubmit', 'feature_admin-additional-actions_avoid_working_via_dbshell', 'feature_dynamic_upload_waiting_time', 'feature_autoattestchecker', 'feature_sort_tasklist_via_names', 'java-checker', 'feature_CUnitCppUnit_Checker', 'feature_LDAP_bindings', 'feature_checkstyleChecker_updVersion_8.14', 'feature_TaskAdmin_Button_Modell_Solution', 'feature_disable_contactlink', 'feature_download_solutiondetails', 'bugfix_safeexec' and 'feature_diffchecker' into hbrs_public

ifrh committed:
[py2py3] fix importing Tasks

Commit:
[py2py3 bugfix] Merge branch 'master32' into java-checker

Commit:
[py2py3 bugfix] Merge all HBRS feature und bugfix branches 'feature_gitmodules_https-proxy_compatible', 'bugfix_small_typos', 'feature_TrainerTutor_Watch_SolutionFiles', 'bugfix_solution_resubmit', 'feature_admin-additional-actions_avoid_working_via_dbshell', 'feature_dynamic_upload_waiting_time', 'feature_autoattestchecker', 'feature_sort_tasklist_via_names', 'java-checker', 'feature_CUnitCppUnit_Checker', 'feature_LDAP_bindings', 'feature_checkstyleChecker_updVersion_8.14', 'feature_TaskAdmin_Button_Modell_Solution', 'feature_disable_contactlink', 'feature_download_solutiondetails', 'bugfix_safeexec' and 'feature_diffchecker' into hbrs_public

Commit:
Merge branch 'master32' after updating from KIT-master into hbrs_public

Conflicts:
	README.md
	src/checker/checker/__init__.py
	src/checker/compiler/Builder.py
	src/settings/defaults.py
	src/settings/devel.py
	src/solutions/views.py

Commit:
remove trailing whitespaces

Commit:
Merge branch 'hbrs_changes_dont_merge_just_for_reference' into 'hbrs_public'
merge of old "hbrs_change" into new "hbrs_public" is possible because of revert like commit de5ab7f6 : there is no file change set.

See merge request rhartm2m/praktomat_hbrs!27

hannesbraun committed:
Move logic to check "latest only failed" solutions to models

hannesbraun committed:
GitHub Actions: also run taskstatistics tests

hannesbraun committed:
GitHub Actions: remove gcj-jdk comment

hannesbraun committed:
Fix solution upload permissions for trainers

hannesbraun committed:
Add runallcheckers command

hannesbraun committed:
Document automated execution of checkers in README

Commit:
Merge branch 'master'from repository KITPraktomatTeam into hbrs_public (15.8.2022)

Conflicts:
	README.md
	requirements.txt
	src/accounts/ldap_auth.py
	src/checker/checker/CUnitChecker_v2.py
	src/checker/checker/__init__.py
	src/checker/compiler/Builder.py
	src/checker/compiler/CLinker.py
	src/checker/migrations/0013_javachecker.py
	src/settings/defaults.py
	src/solutions/views.py
	src/tasks/admin.py
	src/utilities/safeexec.py

Commit:
Merge remote-tracking branches 'hannesbraun/checker-automation', 'hannesbraun/fix-trainer-upload-permission' and 'hannesbraun/update-gh-actions' into hbrs_public

Commit:
README update: Mentione system dependencies introduced via merged feature boxplot-taskstatistic.
fix some orthography mistakes
adding missed migration files

Commit:
Update README.md: Insert information about new dependencies introduced by merging "feature boxplot-taskstatistic" ( https://github.com/KITPraktomatTeam/Praktomat/pull/345 )  .
Update README.md to give more information about installation steps : Fixes https://github.com/KITPraktomatTeam/Praktomat/issues/250 .

Include change sets from https://github.com/KITPraktomatTeam/Praktomat/pull/351 , https://github.com/KITPraktomatTeam/Praktomat/pull/349 , https://github.com/KITPraktomatTeam/Praktomat/pull/348

ifrh committed:
Update README.md

... forgotten to correct path-information in section where "populating the database" is described via running "manage-local.py" .

hannesbraun commented on src/checker/basemodels.py:
```diff
@@ -420,8 +420,10 @@ def run_checks(solution, env, run_all):
+                # TODO: Move code to checker class ?
                 result = checker.create_result(env)
-                result.set_log("Checker konnte nicht ausgeführt werden, da benötigte Checker nicht bestanden wurden.")
+                #result.set_log("Checker konnte nicht ausgeführt werden, da benötigte Checker nicht bestanden wurden.")
```
```suggestion
```

This comment isn't required any longer, isn't it?

ifrh replied:
There exists an issue covering an idea of internationalising messanges : https://github.com/KITPraktomatTeam/Praktomat/issues/215 : I thought not to loose the "original german" message is a nice idea, so I kept it as comment.

hannesbraun commented on src/checker/tests.py:
```diff
@@ -187,7 +187,8 @@ def test_script_timeout(self):
                 self.assertIn('1', checkerresult.log, "Test did not even start?")
-                self.assertIn('Timeout occured!', checkerresult.log, "Test result does not mention timeout")
+                self.assertIn('Timeout occurred!', checkerresult.log, "Test result does not mention timeout")
+                # TODO: 'occured vs occurred' ? (string defined in checker/basemodels.py : CheckerResult.set_log)
```
```suggestion
```

The error message was adjusted accordingly in `checker/basemodels.py`. From what I can see, this TODO comment is obsolete. Or am I missing something?

ifrh replied:
Yes, the TODO followed by the question could be removed. But the pointer where the string has been defined I would like to keep.

ifrh replied:
changed via https://github.com/KITPraktomatTeam/Praktomat/pull/352/commits/5b0d3ef4f5430b6471c8d1b24a51a41515a3888f

hannesbraun commented on src/checker/tests.py:
```diff
@@ -209,7 +210,8 @@ def test_script_filesizelimit(self):
                 self.assertNotIn('End', checkerresult.log, "Test did finish (no timeout?) (%s)" % checkerresult.log)
-                #self.assertIn('Timeout occured!', checkerresult.log, "Test result does not mention timeout")
+                #self.assertIn('Timeout occurred!', checkerresult.log, "Test result does not mention timeout")
+                # TODO: 'occured vs occurred' ? (string defined in checker/basemodels.py : CheckerResult.set_log)
```
```suggestion
```

Same as above

ifrh replied:
:-)

ifrh replied:
changed via https://github.com/KITPraktomatTeam/Praktomat/pull/352/commits/5b0d3ef4f5430b6471c8d1b24a51a41515a3888f

hannesbraun commented on src/checker/tests.py:
```diff
@@ -229,7 +231,8 @@ def test_script_memorylimit(self):
                 self.assertNotIn('End', checkerresult.log, "Test did finish (no timeout?)")
-                self.assertIn('Timeout occured!', checkerresult.log, "Test result does not mention timeout")
+                self.assertIn('Timeout occurred!', checkerresult.log, "Test result does not mention timeout")
+                # TODO: 'occured vs occurred' ? (string defined in checker/basemodels.py : CheckerResult.set_log)
```
```suggestion
```

Same as above

ifrh replied:
yes, the same as above.

ifrh replied:
changed via https://github.com/KITPraktomatTeam/Praktomat/pull/352/commits/5b0d3ef4f5430b6471c8d1b24a51a41515a3888f

hannesbraun commented on README.md:
```diff
@@ -83,13 +84,86 @@ Pitfalls while Systemupgrades
+    django.db.utils.OperationalError: near "(": syntax error while migrating migrationfile taskstatistics.0001_initial_TaskStatisticsDBview
+
+
+In example `SQLite3` version 3.16.2 dated to 2017-01-06 is much to old to handle the hand written migration file `Praktomat/src/taskstatistics/migrations/0001_initial_TaskStatisticsDBview.py` correctly.
+
+Since that migration file use *SQL window functions*, which where standarized in `SQL:2003` and covered by `SQLite3` with version 3.25.0 dated to 2018-09-15. But, because of some bugs inside `SQLite3`, you cannot use that first version supporting *SQL window functions*, too.
```
```suggestion
In example, `SQLite3` version 3.16.2 dated to 2017-01-06 is much too old to handle the handwritten migration file `Praktomat/src/taskstatistics/migrations/0001_initial_TaskStatisticsDBview.py` correctly.

That migration file uses *SQL window functions*, which were standardized in `SQL:2003` and covered by `SQLite3` with version 3.25.0 dated to 2018-09-15. But, because of some bugs inside `SQLite3`, you cannot use that first version supporting *SQL window functions*, too.
```

When we're about to fix orthography errors... ;)

hannesbraun commented on README.md:
```diff
@@ -83,13 +84,86 @@ Pitfalls while Systemupgrades
+
+    apt install tk tk-dev
+
+**Hind:** Praktomat's DejaGnuChecker uses `dejagnu`, which is written in `expect`. And `expect` is using `Tcl`.
```
```suggestion
**Hint:** Praktomat's DejaGnuChecker uses `dejagnu`, which is written in `expect`. And `expect` is using `Tcl`.
```

hannesbraun commented on README.md:
```diff
@@ -83,13 +84,86 @@ Pitfalls while Systemupgrades
+
+ That commands should not produce any error messages.
+
+ If `lzma` couldn't be imported, than you need to install the system library and dev package:
+
+    apt-get install liblzma-dev lzma -y -q
+
+ If `_tkinter` failed to load, than perhaps on your system the `_tkinter` module can be installed for your Python on a seperate way.
+ In example on Debian you install the tkinter-support via system package manager:
```
```suggestion
 If `lzma` couldn't be imported, then you need to install the system library and dev package:

    apt-get install liblzma-dev lzma -y -q

 If `_tkinter` failed to load, then perhaps on your system the `_tkinter` module can be installed for your Python on a separate way.
 In example, on Debian you install the tkinter-support via system package manager:
```

hannesbraun reviewed (changes requested):
I noticed some left over comments and typos. Nothing big though. Otherwise, this looks good :)

ifrh committed:
Update README.md - hannesbraun found some orthography errors.

Co-authored-by: Hannes Braun <hannesbraun@mail.de>

ifrh committed:
Update README.md - hannesbraun found some more orthography errors.

Co-authored-by: Hannes Braun <hannesbraun@mail.de>

ifrh committed:
Update README.md - again hannesbraun found orthography errors.

Co-authored-by: Hannes Braun <hannesbraun@mail.de>

ifrh committed:
Update README.md - fix "to old" => "too old"

ifrh committed:
Update src/checker/tests.py : remove some TODO comments, which were obsolete : found by hannesbraun

hannesbraun reviewed (approved):
Great :)

Btw, sorry for the occasional nitpicking

ifrh committed:
Update local.py

befor match.group('tba') could asked, match has not to be None.

ifrh committed:
Update README.md : mention that policy files could be changed

ifrh committed:
Bugfix in src/tasks/admin.py : prevent Error 500

caling method run_all_uploadtime_checkers_on_all when there is only one solution generates an Error 500.

ifrh committed:
Update apache_praktomat_wsgi.conf

a third party C extension module for Python has been used which will not run correctly within any additional sub interpreters created by Python. So here we change configuration to force our WSGI application to run within the first interpreter. The consequence is: Any WSGI applications in the global application group will always be executed within the context of the first interpreter created by Python when it is initialised, of the process handling the request. See ISSUE https://github.com/KITPraktomatTeam/Praktomat/issues/353

ifrh commented:
@ratefuchs : This pull-request contains additionaly commit https://github.com/ifrh/Praktomat/commit/67453afa87be1d5bd02e425bd6b8124fd817bbbb 
With that commit the apache macro `Praktomat/documentation/apache_praktomat_wsgi.conf` did not produce 
**Fatal Python error : _PyInterpreterState_DeleteExceptMain: not main interpreter**
while running `pool.map(check_it, solutions, 1)` in `check_muliple` from `checker.basemodels` , mentioned in ISSUE https://github.com/KITPraktomatTeam/Praktomat/issues/353

__
Edit: 15.09.2022 : correct path information in this message.

ifrh commented:
This fixes issue https://github.com/KITPraktomatTeam/Praktomat/issues/353

physikerwelt reviewed (commented):
At a first glance it looks good. However, this is a massive pull requests. I can't formally review it. Maybe it would make sense to have one PR for typos and one for each feature?
